django/db/backends/postgresql/operations.py

```diff
@@ -257,3 +258,18 @@ def window_frame_range_start_end(self, start=None, end=None):
                 'and FOLLOWING.'
             )
         return start_, end_
+
+    def explain_query_prefix(self, output_format=None, verbose=False):
+        prefix = super().explain_query_prefix(output_format, verbose)
+
+        extra = {}
```

Comment:
Following on from https://github.com/django/django/pull/9053#discussion_r143178055 which didn't seem to have a response...

PostgreSQL has the following additional options in all supported versions: `BUFFERS`, `COSTS`, `TIMING`.
Version 10+ also supports an additional `SUMMARY` option. (See [documentation](https://www.postgresql.org/docs/10/static/sql-explain.html) for more details.)

Note that some of these options are already enabled by default but can be explicitly disabled. It could be easier to just enable all of these options when verbose is set if it doesn't hugely increase the time taken to execute the statement.

Reply:
Oh, I must have forgotten to respond to that comment. I didn't want to go too deep into adding support for all the various flags, but I think that I will after reading your comments. I'll let `.explain()` accept arbitrary keyword arguments and the various `explain_query_prefix` functions can do what they want with them.

Reply:
Yeah. There are quite a few, and this is anything but standard across backends. The advantage is being able to access all of this information without needing to pop out to the database shell and restricting the flags may cripple the use of this feature for some.

Adding support via kwargs sounds sensible. The irritation may be determining the default state of the various flags based on the value of `verbose`. So it may be easier to keep the interface simple and just "enable all the things" if `verbose` is `True`, where they are not costly in terms of execution time, and then provide `analyze` separately in kwargs.